core/geometry/src/test/geometry3d/Matrix3d.test.ts

```diff
@@ -18,160 +18,172 @@ import * as bsiChecker from "../Checker";
+  const matrix = Matrix3d.createRotationAroundVector(Vector3d.create(0, 0, 1), Angle.createDegrees(45.0));
+  if (matrix) {
+    const vectorA = Vector3d.create(10, 1, 1);
+    const vectorB = Vector3d.create(2, 5, 4);
+    ck.testVector3d(
+      matrix.multiplyVector(vectorA),
+      Vector3d.create(0, 0, 0).plus3Scaled(
+        matrix.columnX(), vectorA.x,
+        matrix.columnY(), vectorA.y,
+        matrix.columnZ(), vectorA.z
+      ),
+      "matrix * vector versus column * scalar"
+    );
+    ck.testVector3d(
+      matrix.multiplyVector(vectorA),
+      Vector3d.create(0, 0, 0).plus2Scaled(
+        matrix.columnY(), vectorA.y,
+        matrix.columnZ(), vectorA.z
+      ).plusScaled(matrix.columnX(), vectorA.x),
+      "matrix * vector versus column * scalar"
+    );
+
+    const frame = Matrix3d.createRigidFromColumns(vectorA, vectorB, AxisOrder.XYZ);
+    if (ck.testPointer(frame, "PerpendicularColumns") && frame
+      && ck.testBoolean(true, frame.testPerpendicularUnitRowsAndColumns(), "UnitPerpendicularColumns")) {
+      const matrixT = matrix.transpose();
+      const frameT = frame.transpose();
+      for (let i = 0; i < 6; i++) {
+        ck.testVector3d(
+          matrix.getRow(i),
+          matrixT.getColumn(i),
+          "row and column match in matrix and its transpose"
+        );
+      }
+      ck.testMatrix3d(
+        matrixT.multiplyMatrixMatrix(frame),
+        matrix.multiplyMatrixTransposeMatrix(frame),
+        "multiplyMatrixTransposeMatrix"
+      );
+      ck.testMatrix3d(
+        matrix.multiplyMatrixMatrix(frameT),
+        matrix.multiplyMatrixMatrixTranspose(frame),
+        "multiplyMatrixMatrixTranspose"
+      );
+      ck.testPerpendicular(vectorA, frame.columnZ(), "vectorA perp frame.Z");
+      ck.testPerpendicular(vectorB, frame.columnZ(), "vectorB perp frame.Z");
+      ck.testParallel(vectorA, frame.columnX(), "vectorA parallel frame.X");
+      ck.testCoordinateOrder(
+        0,
+        vectorB.dotProduct(frame.columnY()),
+        "vectorB in positive XY half plane (halved by X toward Y)"
+      );
+      MatrixTests.checkProperties(ck, frame, false, true, true, true, undefined);
+
+      const frame1 = frame.multiplyMatrixMatrix(Matrix3d.createScale(1, 1, 2));
+      // frame1 still has perpendicular columns but not unit. Also its determinant is not +1 so it's not rigid.
+      MatrixTests.checkProperties(ck, frame1, false, false, false, true, false);
+      // frame2 still has unit perpendicular columns but its determinant is not +1 so it's not rigid.
+      const frame2 = frame.multiplyMatrixMatrix(Matrix3d.createScale(1, 1, -1));
+      MatrixTests.checkProperties(ck, frame2, false, true, false, true, false);
+
+      let vector;
+      const e = 1.0 / 64.0;
+      for (vector of [
+        Vector3d.create(1, 2, 3),
+        Vector3d.create(1, 0, 0),
+        Vector3d.create(0, 1, 0),
+        Vector3d.create(0, 0, 1),
+        Vector3d.create(e, e, 3.0), // triggers near-z logic
```

Comment:
Looks like `Vector3d.create(e, e, 0.5)` test was lost.

Reply:
I intentionally removed it as it does serve the same purpose as (e, e, 3.0)